### PR TITLE
feat: Add support for `sign` function in pgsql

### PR DESCRIPTION
**Pull request by Guo-Shiyu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Add support for `sign` function in pgsql.      
[associated issue: #8900](https://github.com/risingwavelabs/risingwave/issues/8900)

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

- SQL commands, functions, and operators

### Release note

</details>

### CODE REVIEW

**Comment by Guo-Shiyu** [comment]
It's not ready to be merged now and I will submit the test in subsequent commits.  I opened this so shat we can discuss  incorrect code changes here.        

This is my first PR, so let me know if anything looks wrong.  Thanks :)

**Comment by Guo-Shiyu** [comment]
I think it's ready to be merged.  What should I do next ?

**Comment by lmatz** [comment]
Hi @Guo-Shiyu , thanks for your interest in RisingWave and your contribution!

Could you also add the `sign` function for `decimal/numeric` as PostgreSQL supports that and RisingWave tries to be compatible with PostgreSQL?
https://www.postgresql.org/docs/9.1/functions-math.html
SCR-20230515-gef

Could you also add some e2e tests for the function? Thanks you

File: src/expr/src/vector_op/arithmetic_op.rs [hunk 1]
```diff
@@ -405,6 +405,15 @@ pub fn sqrt_decimal(expr: Decimal) -> Result<Decimal> {
     }
 }
 
+#[function("sign(int16) -> int16")]
+#[function("sign(int32) -> int32")]
+#[function("sign(int64) -> int64")]
+#[function("sign(float32) -> float32")]
```
[the remark this answers is not in the document]
Comment (xiangjinwu) [reply src/expr/src/vector_op/arithmetic_op.rs:411]: In addition:
* `signum` behavior is different from PostgreSQL for `0`, `-0` and `NaN`
* our `Decimal` wrapper type does not `impl Signed` (while the inner `RustDecimal` does)
* PostgreSQL `sign('nan'::Typ)` is different when `Typ` is `decimal` or `double precision`

Given these, it is unnecessary to do a `general_sign` with `Signed` trait. Just write 2 functions, one for `decimal` and one for `double precision`.

**Comment by fuyufjh** [comment]
Ping @Guo-Shiyu. The PR has been inactive for a week. Hope it can be merged soon.

**Comment by xiangjinwu** [comment]
superseded by #10819

**Pull request closed without merging**